noir-contracts/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM 278380418400.dkr.ecr.eu-west-2.amazonaws.com/noir as noir
+FROM 278380418400.dkr.ecr.eu-west-2.amazonaws.com/aztec-nr as aztec-nr
+FROM 278380418400.dkr.ecr.eu-west-2.amazonaws.com/noir-protocol-circuits as noir-protocol-circuits
+FROM 278380418400.dkr.ecr.eu-west-2.amazonaws.com/avm-transpiler as transpiler
+FROM node:18.19.0
+
+# Copy in nargo
+COPY --from=noir /usr/src/noir/target/release/nargo /usr/src/noir/target/release/nargo
+# Copy in aztec-nr
+COPY --from=aztec-nr /usr/src/aztec-nr /usr/src/aztec-nr
+# Copy in noir-protocol-circuits
+COPY --from=noir-protocol-circuits /usr/src/noir-protocol-circuits /usr/src/noir-protocol-circuits
+# Copy in transpiler
+COPY --from=transpiler /usr/src/avm-transpiler/target/release/avm-transpiler /usr/src/avm-transpiler/target/release/avm-transpiler
+
+WORKDIR /usr/src/noir-contracts
+# Copy in noir contracts
+COPY . .
+# Build
+RUN yarn build
```

Comment:
I think we could run `compile.sh` and `transpile.sh` directly here then we could remove `FROM node` at the top and reduce the image size further?

Reply:
Did consider that, I just don't like having scripts repeated in multiple places as they can easily be forgotten (someone updates `package.json` but not the Dockerfile)

Reply:
updated 👍